src/main/java/com/amazon/opendistroforelasticsearch/knn/index/KNNVectorFieldMapper.java

```diff
@@ -344,6 +312,12 @@ public void parse(ParseContext context) throws IOException {
             array[i++] = f;
         }
 
+        FieldType fieldType = new FieldType(Defaults.FIELD_TYPE);
+        fieldType.putAttribute(KNNConstants.SPACE_TYPE, spaceType);
+        fieldType.putAttribute(KNNConstants.HNSW_ALGO_M, String.valueOf(m));
+        fieldType.putAttribute(KNNConstants.HNSW_ALGO_EF_CONSTRUCTION, String.valueOf(efConstruction));
```

Comment:
How about reading these algo params from the indexSettings here? Idea is to lazy load these settings to ensure the settings are available to read from IndexSettings.

Reply:
This would cause index settings to be read everytime a document is indexed. I refactored to build fieldType to the KNNVectorFieldMapper constructor and read the settings from BuilderContext during Builder.build()